Deal with Ubuntu 20.04 warnings on clean compile

Compile mantid project using the 20.04 getting started instructions: http://developer.mantidproject.org/GettingStarted.html

Find any warnings and fix the code so we don't get the warnings.

Below is the build output on Ubuntu 20.04
```
$ ninja
[71/2892] Building CXX object Framework/Kernel/CMakeFiles/Kernel.dir/src/NexusHDF5Descriptor.cpp.o
In file included from /usr/include/string.h:495,
                 from /usr/include/c++/9/cstring:42,
                 from /home/svh/Documents/MantidProject/mantid/Framework/Kernel/src/NexusHDF5Descriptor.cpp:17:
In function ‘char* strncpy(char*, const char*, size_t)’,
    inlined from ‘herr_t Mantid::Kernel::{anonymous}::readStringAttributeN(hid_t, char*, int)’ at /home/svh/Documents/MantidProject/mantid/Framework/Kernel/src/NexusHDF5Descriptor.cpp:108:12,
    inlined from ‘Mantid::Kernel::{anonymous}::getGroup(hid_t, std::map<std::__cxx11::basic_string<char>, std::set<std::__cxx11::basic_string<char> > >&)::<lambda(hid_t, const char*)>’ at /home/svh/Documents/MantidProject/mantid/Framework/Kernel/src/NexusHDF5Descriptor.cpp:135:25,
    inlined from ‘void Mantid::Kernel::{anonymous}::getGroup(hid_t, std::map<std::__cxx11::basic_string<char>, std::set<std::__cxx11::basic_string<char> > >&)’ at /home/svh/Documents/MantidProject/mantid/Framework/Kernel/src/NexusHDF5Descriptor.cpp:157:65:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:106:34: warning: ‘char* __builtin_strncpy(char*, const char*, long unsigned int)’ specified bound 128 equals destination size [-Wstringop-truncation]
  106 |   return __builtin___strncpy_chk (__dest, __src, __len, __bos (__dest));
      |          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[539/2892] Building CXX object Framework/API/CMakeFiles/API.dir/src/ScriptRepository.cpp.o
/home/svh/Documents/MantidProject/mantid/Framework/API/src/ScriptRepository.cpp: In constructor ‘Mantid::API::ScriptRepoException::ScriptRepoException(int, const string&, const string&, int)’:
/home/svh/Documents/MantidProject/mantid/Framework/API/src/ScriptRepository.cpp:33:27: warning: ‘%d’ directive output may be truncated writing between 1 and 10 bytes into a region of size 9 [-Wformat-truncation=]
   33 |       snprintf(aux, 10, ":%d", line);
      |                           ^~
/home/svh/Documents/MantidProject/mantid/Framework/API/src/ScriptRepository.cpp:33:25: note: directive argument in the range [1, 2147483647]
   33 |       snprintf(aux, 10, ":%d", line);
      |                         ^~~~~
In file included from /usr/include/stdio.h:867,
                 from /usr/include/c++/9/cstdio:42,
                 from /usr/include/c++/9/ext/string_conversions.h:43,
                 from /usr/include/c++/9/bits/basic_string.h:6493,
                 from /usr/include/c++/9/string:55,
                 from /home/svh/Documents/MantidProject/mantid/Framework/API/inc/MantidAPI/ScriptRepository.h:9,
                 from /home/svh/Documents/MantidProject/mantid/Framework/API/src/ScriptRepository.cpp:7:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:67:35: note: ‘__builtin___snprintf_chk’ output between 3 and 12 bytes into a destination of size 10
   67 |   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
      |          ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   68 |        __bos (__s), __fmt, __va_arg_pack ());
      |        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/svh/Documents/MantidProject/mantid/Framework/API/src/ScriptRepository.cpp: In constructor ‘Mantid::API::ScriptRepoException::ScriptRepoException(const string&, const string&, const string&, int)’:
/home/svh/Documents/MantidProject/mantid/Framework/API/src/ScriptRepository.cpp:51:27: warning: ‘%d’ directive output may be truncated writing between 1 and 10 bytes into a region of size 9 [-Wformat-truncation=]
   51 |       snprintf(aux, 10, ":%d", line);
      |                           ^~
/home/svh/Documents/MantidProject/mantid/Framework/API/src/ScriptRepository.cpp:51:25: note: directive argument in the range [1, 2147483647]
   51 |       snprintf(aux, 10, ":%d", line);
      |                         ^~~~~
In file included from /usr/include/stdio.h:867,
                 from /usr/include/c++/9/cstdio:42,
                 from /usr/include/c++/9/ext/string_conversions.h:43,
                 from /usr/include/c++/9/bits/basic_string.h:6493,
                 from /usr/include/c++/9/string:55,
                 from /home/svh/Documents/MantidProject/mantid/Framework/API/inc/MantidAPI/ScriptRepository.h:9,
                 from /home/svh/Documents/MantidProject/mantid/Framework/API/src/ScriptRepository.cpp:7:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:67:35: note: ‘__builtin___snprintf_chk’ output between 3 and 12 bytes into a destination of size 10
   67 |   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
      |          ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   68 |        __bos (__s), __fmt, __va_arg_pack ());
      |        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[833/2892] Building CXX object Framework/DataHandling/CMakeFiles/DataHandling.dir/src/FilterEventsByLogValuePreNexus.cpp.o
In file included from /usr/include/boost/config/header_deprecated.hpp:18,
                 from /usr/include/boost/timer.hpp:20,
                 from /home/svh/Documents/MantidProject/mantid/Framework/DataHandling/src/FilterEventsByLogValuePreNexus.cpp:37:
/usr/include/boost/timer.hpp:21:1: note: #pragma message: This header is deprecated. Use the facilities in <boost/timer/timer.hpp> instead.
   21 | BOOST_HEADER_DEPRECATED( "the facilities in <boost/timer/timer.hpp>" )
      | ^~~~~~~~~~~~~~~~~~~~~~~
[849/2892] Building CXX object Framework/DataHandling/CMakeFiles/DataHandling.dir/src/LoadEventPreNexus2.cpp.o
In file included from /usr/include/boost/config/header_deprecated.hpp:18,
                 from /usr/include/boost/timer.hpp:20,
                 from /home/svh/Documents/MantidProject/mantid/Framework/DataHandling/src/LoadEventPreNexus2.cpp:43:
/usr/include/boost/timer.hpp:21:1: note: #pragma message: This header is deprecated. Use the facilities in <boost/timer/timer.hpp> instead.
   21 | BOOST_HEADER_DEPRECATED( "the facilities in <boost/timer/timer.hpp>" )
      | ^~~~~~~~~~~~~~~~~~~~~~~
[1580/2892] Generating ../../bin/mantid.egg-link
running develop
running egg_info
creating mantid.egg-info
writing mantid.egg-info/PKG-INFO
writing dependency_links to mantid.egg-info/dependency_links.txt
writing top-level names to mantid.egg-info/top_level.txt
writing manifest file 'mantid.egg-info/SOURCES.txt'
reading manifest file 'mantid.egg-info/SOURCES.txt'
writing manifest file 'mantid.egg-info/SOURCES.txt'
running build_ext
Creating /home/svh/Documents/MantidProject/build/bin/site.py
Creating /home/svh/Documents/MantidProject/build/bin/mantid.egg-link (link to .)
Adding mantid 5.1.1 to easy-install.pth file

Installed /home/svh/Documents/MantidProject/mantid/Framework/PythonInterface
Processing dependencies for mantid==5.1.1
Finished processing dependencies for mantid==5.1.1
[1639/2892] Building CXX object Framework/ICat/CMakeFiles/ICat.dir/src/GSoap.cpp.o
In file included from /usr/include/string.h:495,
                 from /home/svh/Documents/MantidProject/mantid/Framework/ICat/inc/MantidICat/GSoap/stdsoap2.h:620,
                 from /home/svh/Documents/MantidProject/mantid/Framework/ICat/inc/MantidICat/GSoap/soapserializersStub.h:18,
                 from /home/svh/Documents/MantidProject/mantid/Framework/ICat/inc/MantidICat/GSoap/soapserializersH.h:18,
                 from /home/svh/Documents/MantidProject/mantid/Framework/ICat/src/GSoap/soapserializersC.cpp:22,
                 from /home/svh/Documents/MantidProject/mantid/Framework/ICat/src/GSoap.cpp:26:
In function ‘char* strncpy(char*, const char*, size_t)’,
    inlined from ‘wchar_t* soap_wstring_in(soap*, int, long int, long int)’ at /home/svh/Documents/MantidProject/mantid/Framework/ICat/src/GSoap/stdsoap2.cpp:11572:14:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:106:34: warning: ‘char* __builtin_strncpy(char*, const char*, long unsigned int)’ output may be truncated copying 1022 bytes from a string of length 1023 [-Wstringop-truncation]
  106 |   return __builtin___strncpy_chk (__dest, __src, __len, __bos (__dest));
      |          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In function ‘char* strncpy(char*, const char*, size_t)’,
    inlined from ‘char* soap_string_in(soap*, int, long int, long int)’ at /home/svh/Documents/MantidProject/mantid/Framework/ICat/src/GSoap/stdsoap2.cpp:11003:12:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:106:34: warning: ‘char* __builtin_strncpy(char*, const char*, long unsigned int)’ output may be truncated copying 1022 bytes from a string of length 1023 [-Wstringop-truncation]
  106 |   return __builtin___strncpy_chk (__dest, __src, __len, __bos (__dest));
      |          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In function ‘char* strcat(char*, const char*)’,
    inlined from ‘int http_parse(soap*)’ at /home/svh/Documents/MantidProject/mantid/Framework/ICat/src/GSoap/stdsoap2.cpp:5514:13:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:128:33: warning: ‘char* __builtin___strcat_chk(char*, const char*, long unsigned int)’ accessing 1025 or more bytes at offsets 124472 and 125496 may overlap 1 byte at offset 125496 [-Wrestrict]
  128 |   return __builtin___strcat_chk (__dest, __src, __bos (__dest));
      |          ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[1727/2892] Building CXX object Framework/Muon/CMakeFiles/Muon.dir/src/PSIBackgroundSubtraction.cpp.o
/home/svh/Documents/MantidProject/mantid/Framework/Muon/src/PSIBackgroundSubtraction.cpp: In member function ‘virtual std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> > Mantid::Muon::PSIBackgroundSubtraction::validateInputs()’:
/home/svh/Documents/MantidProject/mantid/Framework/Muon/src/PSIBackgroundSubtraction.cpp:92:33: warning: catching polymorphic type ‘class Mantid::Kernel::Exception::NotFoundError’ by value [-Wcatch-value=]
   92 |     } catch (Kernel::Exception::NotFoundError) {
      |                                 ^~~~~~~~~~~~~
/home/svh/Documents/MantidProject/mantid/Framework/Muon/src/PSIBackgroundSubtraction.cpp:99:33: warning: catching polymorphic type ‘class Mantid::Kernel::Exception::NotFoundError’ by value [-Wcatch-value=]
   99 |     } catch (Kernel::Exception::NotFoundError) {
      |                                 ^~~~~~~~~~~~~
[1783/2892] Building CXX object Framework/MDAlgorithms/CMakeFiles/MDAlgorithms.dir/src/ConvToMDSelector.cpp.o
In file included from /home/svh/Documents/MantidProject/mantid/Framework/MDAlgorithms/inc/MantidMDAlgorithms/MDEventTreeBuilder.h:11,
                 from /home/svh/Documents/MantidProject/mantid/Framework/MDAlgorithms/inc/MantidMDAlgorithms/ConvToMDEventsWSIndexing.h:10,
                 from /home/svh/Documents/MantidProject/mantid/Framework/MDAlgorithms/src/ConvToMDSelector.cpp:12:
/usr/include/tbb/task_scheduler_init.h:21:154: note: #pragma message: TBB Warning: tbb/task_scheduler_init.h is deprecated. For details, please see Deprecated Features appendix in the TBB reference manual.
   21 | #pragma message("TBB Warning: tbb/task_scheduler_init.h is deprecated. For details, please see Deprecated Features appendix in the TBB reference manual.")
      |                                                                                                                                                          ^
[1815/2892] Building CXX object Framework/MDAlgorithms/CMakeFiles/MDAlgorithms.dir/src/ConvertToMD.cpp.o
In file included from /home/svh/Documents/MantidProject/mantid/Framework/MDAlgorithms/inc/MantidMDAlgorithms/MDEventTreeBuilder.h:11,
                 from /home/svh/Documents/MantidProject/mantid/Framework/MDAlgorithms/inc/MantidMDAlgorithms/ConvToMDEventsWSIndexing.h:10,
                 from /home/svh/Documents/MantidProject/mantid/Framework/MDAlgorithms/src/ConvertToMD.cpp:33:
/usr/include/tbb/task_scheduler_init.h:21:154: note: #pragma message: TBB Warning: tbb/task_scheduler_init.h is deprecated. For details, please see Deprecated Features appendix in the TBB reference manual.
   21 | #pragma message("TBB Warning: tbb/task_scheduler_init.h is deprecated. For details, please see Deprecated Features appendix in the TBB reference manual.")
      |                                                                                                                                                          ^
[1840/2892] Building CXX object Framework/MDAlgorithms/CMakeFiles/MDAlgorithms.dir/src/SmoothMD.cpp.o
/home/svh/Documents/MantidProject/mantid/Framework/MDAlgorithms/src/SmoothMD.cpp: In member function ‘virtual void Mantid::MDAlgorithms::SmoothMD::exec()’:
/home/svh/Documents/MantidProject/mantid/Framework/MDAlgorithms/src/SmoothMD.cpp:456:40: warning: ‘*((void*)(& optionalWeightingWS)+16).std::__shared_count<>::_M_pi’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  456 |   OptionalIMDHistoWorkspace_const_sptr optionalWeightingWS;
      |                                        ^~~~~~~~~~~~~~~~~~~
/home/svh/Documents/MantidProject/mantid/Framework/MDAlgorithms/src/SmoothMD.cpp:456:40: warning: ‘*((void*)(& optionalWeightingWS)+8).std::__shared_ptr<const Mantid::API::IMDHistoWorkspace, __gnu_cxx::_S_atomic>::_M_ptr’ may be used uninitialized in this function [-Wmaybe-uninitialized]
[1892/2892] Generating qt5/inc/MantidQtWidgets/Common/moc_QSettingsHelper.cpp
/home/svh/Documents/MantidProject/mantid/qt/widgets/common/inc/MantidQtWidgets/Common/QSettingsHelper.h:0: Note: No relevant classes found. No output generated.
[2037/2892] Building CXX object Framework/MDAlgorithms/CMakeFiles/MDAlgorithms.dir/src/ConvToMDEventsWSIndexing.cpp.o
In file included from /home/svh/Documents/MantidProject/mantid/Framework/MDAlgorithms/inc/MantidMDAlgorithms/MDEventTreeBuilder.h:11,
                 from /home/svh/Documents/MantidProject/mantid/Framework/MDAlgorithms/inc/MantidMDAlgorithms/ConvToMDEventsWSIndexing.h:10,
                 from /home/svh/Documents/MantidProject/mantid/Framework/MDAlgorithms/src/ConvToMDEventsWSIndexing.cpp:7:
/usr/include/tbb/task_scheduler_init.h:21:154: note: #pragma message: TBB Warning: tbb/task_scheduler_init.h is deprecated. For details, please see Deprecated Features appendix in the TBB reference manual.
   21 | #pragma message("TBB Warning: tbb/task_scheduler_init.h is deprecated. For details, please see Deprecated Features appendix in the TBB reference manual.")
      |                                                                                                                                                          ^
[2691/2892] Generating qt5/moc_ALFCustomInstrumentModel.cpp
/home/svh/Documents/MantidProject/mantid/qt/scientific_interfaces/Direct/ALFCustomInstrumentModel.h:0: Note: No relevant classes found. No output generated.
[2692/2892] Generating qt5/moc_ALFCustomInstrumentMocks.cpp
/home/svh/Documents/MantidProject/mantid/qt/scientific_interfaces/Direct/ALFCustomInstrumentMocks.h:0: Note: No relevant classes found. No output generated.
[2829/2892] Generating ../../bin/mantidqt.egg-link
running develop
running egg_info
creating mantidqt.egg-info
writing mantidqt.egg-info/PKG-INFO
writing dependency_links to mantidqt.egg-info/dependency_links.txt
writing requirements to mantidqt.egg-info/requires.txt
writing top-level names to mantidqt.egg-info/top_level.txt
writing manifest file 'mantidqt.egg-info/SOURCES.txt'
reading manifest file 'mantidqt.egg-info/SOURCES.txt'
writing manifest file 'mantidqt.egg-info/SOURCES.txt'
running build_ext
Creating /home/svh/Documents/MantidProject/build/bin/mantidqt.egg-link (link to .)
Adding mantidqt 5.1.1 to easy-install.pth file

Installed /home/svh/Documents/MantidProject/mantid/qt/python
Processing dependencies for mantidqt==5.1.1
Searching for mantid==5.1.1
Best match: mantid 5.1.1
mantid 5.1.1 is already the active version in easy-install.pth

Using /home/svh/Documents/MantidProject/mantid/Framework/PythonInterface
Finished processing dependencies for mantidqt==5.1.1
[2854/2892] Generating ../../../bin/MantidWorkbench.egg-link, ../../../bin/workbench
running develop
running egg_info
creating MantidWorkbench.egg-info
writing MantidWorkbench.egg-info/PKG-INFO
writing dependency_links to MantidWorkbench.egg-info/dependency_links.txt
writing entry points to MantidWorkbench.egg-info/entry_points.txt
writing requirements to MantidWorkbench.egg-info/requires.txt
writing top-level names to MantidWorkbench.egg-info/top_level.txt
writing manifest file 'MantidWorkbench.egg-info/SOURCES.txt'
reading manifest file 'MantidWorkbench.egg-info/SOURCES.txt'
writing manifest file 'MantidWorkbench.egg-info/SOURCES.txt'
running build_ext
Creating /home/svh/Documents/MantidProject/build/bin/MantidWorkbench.egg-link (link to .)
Adding MantidWorkbench 5.1.1 to easy-install.pth file
Installing workbench script to /home/svh/Documents/MantidProject/build/bin/.

Installed /home/svh/Documents/MantidProject/mantid/qt/applications/workbench
Processing dependencies for MantidWorkbench==5.1.1
Searching for mantidqt==5.1.1
Best match: mantidqt 5.1.1
mantidqt 5.1.1 is already the active version in easy-install.pth

Using /home/svh/Documents/MantidProject/mantid/qt/python
Searching for mantid==5.1.1
Best match: mantid 5.1.1
mantid 5.1.1 is already the active version in easy-install.pth

Using /home/svh/Documents/MantidProject/mantid/Framework/PythonInterface
Finished processing dependencies for MantidWorkbench==5.1.1
[2892/2892] Linking CXX shared library bin/libMantidScientificInterfacesIndirectQt5.so
```

--- a/Framework/DataHandling/inc/MantidDataHandling/LoadEventNexus.h
+++ b/Framework/DataHandling/inc/MantidDataHandling/LoadEventNexus.h
@@ -544,7 +544,7 @@ bool LoadEventNexus::runLoadInstrument(const std::string &nexusfilename, T local
     try {
       instFilename =
           API::InstrumentFileFinder::getInstrumentFilename(instrument, localWorkspace->getWorkspaceStartDate());
-    } catch (Kernel::Exception::NotFoundError) {
+    } catch (Kernel::Exception::NotFoundError &) {
       if (instFilename.empty()) {
         Poco::Path directory(Kernel::ConfigService::Instance().getInstrumentDirectory());
         Poco::Path file(instrument + "_Definition.xml");

